libnetwork/drivers/bridge/bridge_linux.go

```diff
@@ -179,6 +180,34 @@ func Register(r driverapi.Registerer, config map[string]interface{}) error {
 	})
 }
 
+// Linux always assigns link-local addresses from subnet 'fe80::/64'. Allow that
+// subnet to be used, pool-assigned addresses are very unlikely to conflict with
+// a Linux-assigned EUI-64 based address (or the non-EUI-64 MAC based address
+// used by the default bridge). But, don't allow a different LL subnet to overlap
+// with it. For example, if the config asked for subnet 'fe80::/80', the bridge
+// and its containers would each end up with two LL addresses, Linux's '/64' and a
+// pool address in the same subnet but claiming '/80'.
+func validateIPv6Subnet(what string, addr *net.IPNet) error {
+	if addr != nil && bridgeIPv6.Contains(addr.IP) && !bytes.Equal(bridgeIPv6.Mask, addr.Mask) {
+		return &IPv6ReservedLinkLocalError{What: what, Net: addr}
```

Comment:
It looks like `IPv6ReservedLinkLocalError` is only used here, and (from the looks of it), is not used as a sentinel error (possibly "other then implementing `errdefs.ErrInvalidParameter` interface ").

If we don't need it as a sentinel error (i.e., doesn't need specific handling / used in business-logic), I'd like to avoid giving the impression that it's "special";

- at least un-export it, so that we don't expand the public API surface of the package
- if repetition is a concern we _could_ have a more generic `invalidParam{}` struct that implements the `ErrInvalidParameter` interface
- Or use the [`errdefs.InvalidParameter()`](https://github.com/moby/moby/blob/f179243876b98fefaae5a9a4624d92e3eebe38a5/errdefs/helpers.go#L37-L43) utility;


```go
return errdefs.InvalidParameter(fmt.Sprintf("%s clashes with reserved link local subnet 'fe80::/64': %v", what, addr)
```

Reply:
That makes sense, thank you - get rid of the type, and sort it out with `errdefs.InvalidParameter()`.